File: src/main/java/sigmaone/industrialism/energy/IEnergyContainer.java
package sigmaone.industrialism.energy;

public interface IEnergyContainer {
    /*
     * Returns the currently stored energy
     *
     * @return Energy amount
     */
    float getStoredEnergy();
    /*
     * Returns the maximum energy capacity
     *
     * @return Maximum energy amount
     */
    float getMaxEnergy();
    /*
     * Returns the amount of free space in the energy buffer
     *
     * @return Free space
     */
    float getAvailableEnergyCapacity();
}

File: src/main/kotlin/sigmaone/industrialism/energy/IEnergyContainer.kt
package sigmaone.industrialism.energy

interface IEnergyContainer {
    /*
     * Returns the currently stored energy
     *
     * @return Energy amount
     */
    val storedEnergy: Float

    /*
     * Returns the maximum energy capacity
     *
     * @return Maximum energy amount
     */
    val maxEnergy: Float

    /*
     * Returns the amount of free space in the energy buffer
     *
     * @return Free space
     */
    val availableEnergyCapacity: Float
}